[Enhancement] Make region is optional in iceberg rest catalog vended credentials (#53860)
StarRocks/starrocks: docs/en/data_source/catalog/iceberg_catalog.md

###### iceberg.catalog.uri

Required: Yes
Description: The URI of the REST service endpoint. Example: `https://api.tabular.io/ws`. 

###### iceberg.catalog.security

Required: No

> Description: The type of security to use (default: `NONE`). `OAUTH2` requires either a `token` or `credential`. Example: `OAUTH2`.

Reviewer: ```suggestion
Description: The type of authorization protocol to use (default: `NONE`). `OAUTH2` requires either a `token` or `credential`. Example: `OAUTH2`.
```

> Description: The type of security to use (default: `NONE`). `OAUTH2` requires either a `token` or `credential`. Example: `OAUTH2`.

Reviewer: ```suggestion
Description: The type of authorization protocol to use. Default: `NONE`. Valid value: `OAUTH2`, which requires either a `token` or `credential`.
```

###### iceberg.catalog.oauth2.token

Required: No

> Description: The bearer token used for interactions with the server. A `token` or `credential` is required for `OAUTH2` security. Example: `AbCdEf123456`.

Reviewer: ```suggestion
Description: The bearer token used for interactions with the server. A `token` or `credential` is required for `OAUTH2` authorization protocol. Example: `AbCdEf123456`.
```

> Description: The bearer token used for interactions with the server. A `token` or `credential` is required for `OAUTH2` security. Example: `AbCdEf123456`.

Reviewer: ```suggestion
Description: The bearer token used for interactions with the server. A `token` or `credential` is required for `OAUTH2` authorization protocol. Example: `AbCdEf123456`.
```

###### iceberg.catalog.oauth2.credential

> Required: No

Reviewer: ```suggestion
Required: No

```

> Required: No

Reviewer: ```suggestion
Required: No

```

> Description: The credential to exchange for a token in the OAuth2 client credentials flow with the server. A `token` or `credential` is required for `OAUTH2` security. Example: `AbCdEf123456`.

Reviewer: ```suggestion
Description: The credential to exchange for a token in the OAuth2 client credentials flow with the server. A `token` or `credential` is required for `OAUTH2` authorization protocol. Example: `AbCdEf123456`.
```

###### iceberg.catalog.oauth2.scope

Required: No

> Description: Scope to be used when communicating with the REST Catalog. Applicable only when using `credential`.

Reviewer: ```suggestion
Description: Scope to be used when communicating with the REST Catalog. Applicable only when `credential` is used.
```

###### iceberg.catalog.oauth2.server-uri

Required: No

Description: The endpoint to retrieve access token from OAuth2 Server.

###### iceberg.catalog.vended-credentials-enabled

Required: No

> Description: Support querying objects under nested namespace. Defaults to `true`.

Reviewer: ```suggestion
Description: Whether to support querying objects under nested namespace. Default: `true`.
```

###### iceberg.catalog.uri

Required: Yes
Description: The URI of the REST service endpoint. Example: `https://api.tabular.io/ws`. 

###### iceberg.catalog.security

Required: No

Description: The type of authorization protocol to use. Default: `NONE`. Valid value: `OAUTH2`, which requires either a `token` or `credential`.

###### iceberg.catalog.oauth2.token

Required: No

Description: The bearer token used for interactions with the server. A `token` or `credential` is required for `OAUTH2` authorization protocol. Example: `AbCdEf123456`.

###### iceberg.catalog.oauth2.credential

Required: No

> Description: The credential to exchange for a token in the OAuth2 client credentials flow with the server. A `token` or `credential` is required for `OAUTH2` security. Example: `AbCdEf123456`.

Reviewer: ```suggestion
Description: The credential to exchange for a token in the OAuth2 client credentials flow with the server. A `token` or `credential` is required for `OAUTH2` authorization protocol. Example: `AbCdEf123456`.
```

###### iceberg.catalog.oauth2.scope

Required: No

> Description: Scope to be used when communicating with the REST Catalog. Applicable only when using `credential`.

Reviewer: ```suggestion
Description: Scope to be used when communicating with the REST Catalog. Applicable only when `credential` is used.
```

###### iceberg.catalog.oauth2.server-uri

Required: No

Description: The endpoint to retrieve access token from OAuth2 Server.

###### iceberg.catalog.vended-credentials-enabled

Required: No

> Description: Support querying objects under nested namespace. Defaults to `true`.

Reviewer: ```suggestion
Description: Whether to support querying objects under nested namespace. Default: `true`.
```

    "iceberg.catalog.warehouse" = "sandbox"
);
```

> The following example creates an Iceberg catalog named `smith_polaris` that uses Polaris as metastore:

Reviewer: ```suggestion
The following example creates an Iceberg catalog named `iceberg_polaris` that uses Polaris as metastore:
```

```sql

> CREATE EXTERNAL CATALOG smith_polaris 

Reviewer: ```suggestion
CREATE EXTERNAL CATALOG iceberg_polaris 
```

PROPERTIES (   

>     "iceberg.catalog.uri"  = "http://localhost:8181/api/catalog",   

Reviewer: ```suggestion
    "iceberg.catalog.uri"  = "http://xxx.xx.xx.xxx:8181/api/catalog",   
```

    "type"  =  "iceberg",   
    "iceberg.catalog.type"  =  "rest",   
    "iceberg.catalog.warehouse" = "starrocks_catalog",
    "iceberg.catalog.security" = "oauth2",

>     "iceberg.catalog.oauth2.credential" = "f7357cfbe9583aa1:7b90fc77e4fbe288a5c41eaa05da3a3b",

Reviewer: ```suggestion
    "iceberg.catalog.oauth2.credential" = "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
```

    "iceberg.catalog.oauth2.scope"='PRINCIPAL_ROLE:ALL'
 );

# `ns1.ns2.tpch_namespace` is a nested namespace

> create table smith_polaris.`ns1.ns2.tpch_namespace`.tbl (c1 string);

Reviewer: ```suggestion
create table iceberg_polaris.`ns1.ns2.tpch_namespace`.tbl (c1 string);
```

> mysql> select * from smith_polaris.`ns1.ns2.tpch_namespace`.tbl;

Reviewer: ```suggestion
mysql> select * from iceberg_polaris.`ns1.ns2.tpch_namespace`.tbl;
```

    "iceberg.catalog.warehouse" = "sandbox"
);
```
The following example creates an Iceberg catalog named `smith_polaris` that uses Polaris as metastore:

```sql
CREATE EXTERNAL CATALOG smith_polaris 
PROPERTIES (   

>     "iceberg.catalog.uri"  = "http://localhost:8181/api/catalog",   

Reviewer: ```suggestion
    "iceberg.catalog.uri"  = "http://xxx.xx.xx.xxx:8181/api/catalog", 
```

    "type"  =  "iceberg",   
    "iceberg.catalog.type"  =  "rest",   
    "iceberg.catalog.warehouse" = "starrocks_catalog",
    "iceberg.catalog.security" = "oauth2",

>     "iceberg.catalog.oauth2.credential" = "f7357cfbe9583aa1:7b90fc77e4fbe288a5c41eaa05da3a3b",

Reviewer: ```suggestion
    "iceberg.catalog.oauth2.credential" = "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
```
